prefect_kubernetes/flows.py

```diff
@@ -0,0 +1,43 @@
+"""A module to define flows interacting with Kubernetes resources."""
+
+from typing import Any, Dict
+
+from prefect import flow
+
+from prefect_kubernetes.jobs import KubernetesJob
+
+
+@flow
+async def run_namespaced_job(
+    kubernetes_job: KubernetesJob,
+) -> Dict[str, Any]:
+    """Flow for running a namespaced Kubernetes job.
+
+    Args:
+        kubernetes_job: The `KubernetesJob` block that specifies the job to run.
+
+    Returns:
+        The a dict of logs from each pod in the job, e.g. {'pod_name': 'pod_log_str'}.
+
+    Raises:
+        RuntimeError: If the created Kubernetes job attains a failed status.
+
+    Example:
+
+        ```python
+        from prefect_kubernetes import KubernetesJob, run_namespaced_job
+        from prefect_kubernetes.credentials import KubernetesCredentials
+
+        run_namespaced_job(
+            kubernetes_job=KubernetesJob.from_yaml_file(
+                credentials=KubernetesCredentials.load("k8s-creds"),
+                manifest_path="path/to/job.yaml",
+            )
+        )
+        ```
+    """
+    kubernetes_job_run = await kubernetes_job.trigger()
+
+    await kubernetes_job_run.wait_for_completion()
+
+    return await kubernetes_job_run.fetch_result()
```

Comment:
Can we try wrapping these in `task` to get some orchestration in this flow? The suggestion should probably be improved to something more reasonable.
```suggestion
    kubernetes_job_run = await task(kubernetes_job.trigger, retries=3)()

    await task(kubernetes_job_run.wait_for_completion)()

    return await task(kubernetes_job_run.fetch_result)()
```